packages/react-router/modules/Route.js

```diff
@@ -53,10 +49,15 @@ class Route extends React.Component {
     match: this.computeMatch(this.props, this.context.router)
   }
 
-  computeMatch({ computedMatch, location, path, strict, exact }, { route }) {
+  computeMatch({ computedMatch, location, path, strict, exact }, router) {
     if (computedMatch)
       return computedMatch // <Switch> already computed the match for us
 
+    if (!router) {
+      throw new Error('You should not use <Route> or withRoute() outside a valid <Router>')
```

Comment:
`withRouter` not `withRoute`.

Also, shouldn't this be done with `warning` or `invariant`?